signing/signing.go

```diff
@@ -15,6 +15,7 @@ import (
 
 type Claims struct {
 	Addr string `json:"addr"`
+	Hrp  string `json:"hrp"`
```

Comment:
(Has been discussed, but adding here for @benarena sake) - HRP can be derived out of the address (via a split on `address.lastIndexOf('1')`). BIP0137 defines the encoded address as so:
---
 * The human-readable part, which is intended to convey the type of data, or anything else that is relevant to the reader. This part MUST contain 1 to 83 US-ASCII characters, with each character having a value in the range [33-126]. HRP validity may be further restricted by specific applications.
 * The separator, which is always "1". In case "1" is allowed inside the human-readable part, the last one in the string is the separator[3].
 * The data part, which is at least 6 characters long and only consists of alphanumeric characters excluding "1", "b", "i", and "o"[[4]](https://en.bitcoin.it/wiki/BIP_0173#cite_note-4).

Reply:
Updated to use the last index of "1" to find the hrp, throws an error if the separator is not present.